src/wazuh_modules/vulnerability_detector/wm_vuln_detector.c

```diff
@@ -601,9 +615,10 @@ int wm_vuldet_compare_pkg(cve_vuln_pkg *Pkg1, cve_vuln_pkg *Pkg2) {
         return -1;
     }
 
-    if (!(!Pkg1->arch && !Pkg2->arch) &&
-        !(Pkg1->arch && Pkg2->arch &&
-        !strcmp(Pkg1->arch, Pkg2->arch))) { // Architecture
+    if (!Pkg1->arch || !Pkg2->arch) { // Architecture
+        return -1;
+    } else if (strcmp("noarch", Pkg1->arch) &&
+        strcmp("noarch", Pkg2->arch) && strcmp(Pkg1->arch, Pkg2->arch)) {
```

Comment:
Let's suppose for this second condition that `Pkg1->arch` is `x86`, and `Pkg2->arch` is `noarch`, so in the second `strcmp` evaluation, the `if` statement will evaluate as `false` and will continue the execution. Is this the expected behavior? Shouldn't we return `-1` en that case considering the packages as different packages?

Reply:
My bad, I forgot to add a comment explaining that piece of code. This is the expected behaviour. The generic `noarch` vulnerability "contains" the `x86` one since both are identical and affect the same package. So instead of generating duplicate alerts with different architecture, the last one to arrive is treated as a duplicate and discarded.

Reply:
Perfect! You clarified my questions!